File: mmorpg-server/src/main/java/com/common/packetId/impl/ReqLoginPacket.java
package com.common.packetId.impl;


import com.baidu.bjf.remoting.protobuf.annotation.Protobuf;
import com.common.annotation.WsPacket;
import com.common.packetId.AbstractPaket;
import com.common.packetId.PacketId;

//@Component
@WsPacket(packetId = PacketId.LOGIN_REQ)
public class ReqLoginPacket extends AbstractPaket {

    @Protobuf
    private String userName;

    @Protobuf
    private String password;

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Override
    public short getPacketId() {
        return PacketId.LOGIN_REQ;
    }
}

File: mmorpg-browser/mmorpg-browser/src/app/module/packetId/impl/ReqLoginPacket.ts
import { AbstractPacket } from "../AbstractPacket";
import { PacketId } from "../PacketId";

export class ReqLoginPacket extends AbstractPacket{

    private userName : string;
    private password : string;

    /**
     * Getter $userName
     * @return {string}
     */
	public get $userName(): string {
		return this.userName;
	}

    /**
     * Getter $password
     * @return {string}
     */
	public get $password(): string {
		return this.password;
	}

    /**
     * Setter $userName
     * @param {string} value
     */
	public set $userName(value: string) {
		this.userName = value;
	}

    /**
     * Setter $password
     * @param {string} value
     */
	public set $password(value: string) {
		this.password = value;
	}

    public getPacketId():number{
        return PacketId.LOGIN_REQ;
    }
}